libbpf-tools/biostacks.bpf.c

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Wenbo Zhang
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "biostacks.h"
+#include "bits.bpf.h"
+#include "maps.bpf.h"
+
+#define MAX_ENTRIES 10240
+
+const volatile char targ_disk[DISK_NAME_LEN] = {};
+const volatile bool targ_ms = false;
+
+struct internal_rqinfo {
+	u64 start_ts;
+	struct rqinfo rqinfo;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct request *);
+	__type(value, struct internal_rqinfo);
+	__uint(map_flags, BPF_F_NO_PREALLOC);
+} rqinfos SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct rqinfo);
+	__type(value, struct hist);
+	__uint(map_flags, BPF_F_NO_PREALLOC);
+} hists SEC(".maps");
+
+static __always_inline bool disk_filtered(const char *disk)
```

Comment:
nit: filtered is a bit ambiguous (could mean both matched, as well as rejected). disk_allowed() would be more obvious.

Reply:
Agree, thanks.